src/modules/app-manifest.ts

```diff
@@ -23,48 +23,98 @@ const APP_MANIFEST_PATH = resolve(cwd(), DEFAULT_APP_MANIFEST_PATH);
+ * @prop {string} name - Name of the Vehicle App.
+ * @prop {AppManifestInterfaceAttributes[]} interfaces - List of interface entries in the AppManifest.
  */
-export interface AppManifestInterfaces {
-    interfaces: AppManifestInterfaceEntry[];
+export interface AppManifestAttributes {
+    manifestVersion: string;
+    name: string;
+    interfaces: AppManifestInterfaceAttributes[];
 }
 
-export function readAppManifest(appManifestPath: string = APP_MANIFEST_PATH): any | undefined {
-    let manifest: any;
+/**
+ * Represents an AppManifest for a Vehicle App.
+ * @class
+ * @implements {AppManifest}
+ */
+export class AppManifest implements AppManifestAttributes {
+    manifestVersion: string = 'v3';
+    name: string;
+    interfaces: AppManifestInterfaceAttributes[] = [];
+
+    /**
+     * Creates an instance of AppManifest.
+     * @param {string} name - Name of the Vehicle App.
+     * @param {AppManifestInterfaceAttributes[]} [interfaces=[]] - List of interface entries in the AppManifest.
+     */
+    constructor(name: string, interfaces: AppManifestInterfaceAttributes[] = []) {
+        this.name = name;
+        this.interfaces = this._createInterfaceEntries(interfaces);
+    }
 
-    if (existsSync(appManifestPath)) {
-        try {
-            const file = readFileSync(appManifestPath, DEFAULT_BUFFER_ENCODING);
-            manifest = JSON.parse(file);
-            if (manifest instanceof Array) {
-                // for backwards compatibility we use the first entry from the array
-                manifest = manifest[0];
+    /**
+     * Creates interface entries for the AppManifest.
+     * @private
+     * @param {AppManifestInterfaceAttributes[]} interfaces - List of interface entries to process.
+     * @returns {AppManifestInterfaceAttributes[]} Processed interface entries with parsed JSON values.
+     */
+    private _createInterfaceEntries(interfaces: AppManifestInterfaceAttributes[]): AppManifestInterfaceAttributes[] {
+        return interfaces.map((entry: AppManifestInterfaceAttributes) => ({
+            type: entry.type,
+            config: Object.fromEntries(
+                Object.entries(entry.config).map(([key, value]) => {
+                    try {
+                        return [key, JSON.parse(value)];
+                    } catch {
+                        return [key, value];
+                    }
+                }),
+            ),
+        }));
+    }
+
+    /**
+     * Reads an AppManifest from a file.
+     * @static
+     * @param {string} [appManifestPath=APP_MANIFEST_PATH] - Path to the AppManifest file.
+     * @returns {AppManifest|undefined} The AppManifest or undefined if the file doesn't exist.
+     * @throws {Error} Throws an error if there is an issue reading the AppManifest file.
+     */
+    static read(appManifestPath: string = APP_MANIFEST_PATH): AppManifest | undefined {
+        if (existsSync(appManifestPath)) {
```

Comment:
Prefer early returns to reduce the nesting level:

``typescript
if (!existsSync(appManifestPath)) {
    console.info('*** Info ***: No AppManifest found');
    return undefined;
}
``